packages/evm/test/payments/payments-execution.js

```diff
@@ -0,0 +1,144 @@
+      })
+    })
+
+    context('when the court has already started', () => {
+      beforeEach('move terms to reach period #0', async () => {
+        await controller.mockSetTerm(PERIOD_DURATION)
+      })
+
+      context('when the amount is greater than zero', () => {
+        const amount = bigExp(1, 18)
+
+        context('when the sender has enough balance', () => {
+          const from = payer
+
+          context('when paying with tokens', () => {
+            beforeEach('mint tokens', async () => {
+              await token.generateTokens(from, amount)
+              await token.approve(paymentsBook.address, amount, { from })
+            })
+
+            it('pays the requested amount', async () => {
+              const previousPayerBalance = await token.balanceOf(from)
+              const previousPaymentsBookBalance = await token.balanceOf(paymentsBook.address)
+
+              await paymentsBook.pay(token.address, amount, someone, data, { from })
+
+              const currentPayerBalance = await token.balanceOf(from)
+              assertBn(currentPayerBalance, previousPayerBalance.sub(amount), 'payer balances do not match')
+
+              const currentPaymentsBookBalance = await token.balanceOf(paymentsBook.address)
+              assertBn(currentPaymentsBookBalance, previousPaymentsBookBalance.add(amount), 'payments book balances do not match')
+            })
+
+            it('computes the juror and governor fees correctly', async () => {
+              const currentPeriodId = await paymentsBook.getCurrentPeriodId()
+              const { jurorFees: previousJurorFees, governorFees: previousGovernorFees } = await paymentsBook.getPeriodFees(currentPeriodId, token.address)
+
+              await paymentsBook.pay(token.address, amount, someone, data, { from })
+              const { jurorFees: currentJurorFees, governorFees: currentGovernorFees } = await paymentsBook.getPeriodFees(currentPeriodId, token.address)
+
+              const expectedGovernorFees = amount.mul(GOVERNOR_SHARE_PCT).div(PCT_BASE)
+              assertBn(currentGovernorFees, previousGovernorFees.add(expectedGovernorFees), 'period governor fees do not match')
+
+              const expectedJurorFees = amount.sub(expectedGovernorFees)
+              assertBn(currentJurorFees, previousJurorFees.add(expectedJurorFees), 'period juror fees do not match')
+            })
+
+            it('emits an event', async () => {
+              const currentPeriodId = await paymentsBook.getCurrentPeriodId()
+              const receipt = await paymentsBook.pay(token.address, amount, someone, data, { from })
+
+              assertAmountOfEvents(receipt, PAYMENTS_BOOK_EVENTS.PAYMENT_RECEIVED)
+              assertEvent(receipt, PAYMENTS_BOOK_EVENTS.PAYMENT_RECEIVED, { expectedArgs: { periodId: currentPeriodId, payer: someone, sender: from, token, amount } })
+            })
+          })
+
+          context('when paying with ETH', () => {
```

Comment:
As noted in the contracts, it may be worth adding a test to revert if you send ETH but don't use the ETH address (along the account already having an approval for specified token so the `transferFrom()` works).

Reply:
Addressed in b0ee119be0ba5b590fc137bec28df4dfc05bf4b6